modules/ethereum-contract/builtin/src/lib.rs

```diff
@@ -113,16 +131,39 @@ pub fn verify_substrate_finality_proof(
 ) -> Result<(), Error> {
 	let best_set = AuthorityList::decode(&mut &raw_best_set[..])
 		.map_err(Error::BestSetDecode)
-		.and_then(|authorities| VoterSet::new(authorities.into_iter()).ok_or(Error::InvalidBestSet))?;
-	sc_finality_grandpa::GrandpaJustification::<Block>::decode_and_verify_finalizes(
+		.and_then(|authorities| VoterSet::new(authorities.into_iter()).ok_or(Error::InvalidBestSet));
+
+	log::debug!(
+		target: "bridge-builtin",
+		"Parsed Substrate authorities set {}: {:?}",
+		hex::encode(raw_best_set),
+		best_set,
+	);
+
+	let best_set = best_set?;
+
+	let verify_result = sc_finality_grandpa::GrandpaJustification::<Block>::decode_and_verify_finalizes(
 		&raw_finality_proof,
 		(finality_target_hash, finality_target_number),
 		best_set_id,
 		&best_set,
 	)
 	.map_err(Box::new)
 	.map_err(Error::JustificationVerify)
-	.map(|_| ())
+	.map(|_| ());
+
+	log::debug!(
+		target: "bridge-builtin",
+		"Verified Substrate finality proof {}: {:?}",
+		if log::log_enabled!(log::Level::Trace) {
```

Comment:
I wonder if this works correctly, since we override `target` here, looking at the code of `log_enabled` it passes `module_path!()` to the check.

Also I think we might do something similar for authorities as well.

Reply:
OK, it doesn't work - will try to find another way :/ I'd prefer to leave header - it has helped me a lot, so imo it is important to see it even in `debug!()` mode. Will do the same for authorities

Reply:
Ok - I'll change header as well, because it is parsing result that has helped me